OpenRA.Mods.Common/Traits/Production/Production.cs

```diff
@@ -93,27 +95,32 @@ public void DoProduction(Actor self, ActorInfo producee, ExitInfo exitinfo, stri
 				foreach (var t in newUnit.TraitsImplementing<INotifyBuildComplete>())
 					t.BuildingComplete(newUnit);
 			});
+
+			return newUnit;
 		}
 
-		public virtual bool Produce(Actor self, ActorInfo producee, string raceVariant)
+		public virtual bool Produce(Actor self, IEnumerable<ActorInfo> actorsToProduce, string raceVariant)
 		{
 			if (Reservable.IsReserved(self))
 				return false;
 
-			// pick a spawn/exit point pair
-			var exit = self.Info.Traits.WithInterface<ExitInfo>().Shuffle(self.World.SharedRandom)
-				.FirstOrDefault(e => CanUseExit(self, producee, e));
+		    var allProduced = true;
 
-			if (exit != null)
+			foreach (var actorInfo in actorsToProduce)
 			{
-				DoProduction(self, producee, exit, raceVariant);
-				return true;
+				// pick a spawn/exit point pair
+				var exit = self.Info.Traits.WithInterface<ExitInfo>().Shuffle(self.World.SharedRandom)
+					.FirstOrDefault(e => CanUseExit(self, actorInfo, e));
+			    if (exit != null)
+			        DoProduction(self, actorInfo, exit, raceVariant);
+			    else
+			        allProduced = false;
```

Comment:
Again, we really need that StyleCop+ rule, which sadly depends on https://github.com/shuruev/StyleCopPlus/pull/5.